console/v1/types_console_link.go

```diff
@@ -0,0 +1,37 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Extension for customizing OpenShift web console links
+type ConsoleLink struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Link              `protobuf:"bytes,2,opt,name=link"`
+	// menu determines which console menu the link will be appended to (help, user)
+	Menu ConsoleMenuSelector `json:"menu" protobuf:"bytes,3,opt,name=menu,casttype=ConsoleMenuSelector"`
```

Comment:
I wonder if this shouldn't be something more generic like "location" in case we ever want to add links to other places

Reply:
Not a bad idea.

Reply:
updated.